### DISCUSSION TITLE
Multiple triggers in a single App service

### DISCUSSION THREAD

**Discussion by sudhamani-hcl** [edgexfoundry/community#182] [General]
Hi Team,

I need a clarification. Is it possible to configure multiple triggers in a single app service?
If it is possible, can you please point me to any reference example?

Thanks,
Sudhamani

**Comment by lenny-goodell** [comment]
Nope. App SDK is designed around a single trigger. Having multiple triggers would be contrary to proper microservice design of doing one thing and do it good.

**Reply by sudhamani-hcl** [reply]
Hi @lenny-intel,
Thanks for the response. I need to share the data between two app-services.
 
APP1 --> Triggered by HTTP
APP2 --> Triggered by Message bus event

How to share the data between these two APPS?

Thanks,
Sudhamani

**Reply by lenny-goodell** [reply]
Please explain your use case in some more detail. 
What is the purpose of APP1 with the HTTP Trigger?
What is the purpose of APP2 with the Message bus event trigger?
Do you need sharing in both directions or does APP1 only need to send info to APP2 or vise versa.

**Reply by jiekechoo** [reply]
The simplist way, I think, setup 'two sinks' in eKuiper. It depends on your 'application service' , what is your `core purpose`?

**Reply by sudhamani-hcl** [reply]
Hi @lenny-intel ,

Please find the use-case below.

**GUI**
1.Initiate HTTP req

**APP1** 
2. Receives that HTTP request, 
3. Sends a command to the end device to get on-demand data
4. Receives the data from the end device
5. Process and forward the data to APP2

We have a requirement where APP2 needs to take some actions based on the received data from different APPs.

Thanks,
Sudhamani

**Reply by sudhamani-hcl** [reply]
Hi @jiekechoo ,
Thanks for the suggestion. We'll check and let you know.

**Reply by lenny-goodell** [reply]
@sudhamani-hcl , the Command can return the data directly as the response to the request, rather than on the MessageBus. Simply set the Boolean parameters `dsRetrunEvent` to `true` and `dsPushEvent` to `false`.
https://github.com/edgexfoundry/go-mod-core-contracts/blob/v3.0/clients/interfaces/command.go#L28
This will allow you to have a single App Service.

What actions does your App Service need to take "based on the received data"?

**Reply by sudhamani-hcl** [reply]
Thanks @lenny-intel , I will look into this and get back to you.

Thanks,
Sudhamani.
